crates/bevy_ui/src/layout_components.rs

```diff
@@ -0,0 +1,458 @@
+//! Components used to control the layout of UI-[`Node`](super::Node) entities.
+use crate::{Size, UiRect, Val};
+use bevy_derive::{Deref, DerefMut};
+use bevy_ecs::prelude::Component;
+use bevy_reflect::prelude::*;
+use serde::{Deserialize, Serialize};
+
+use self::flex::FlexLayout;
+
+/// Controls which layout algorithm is used to position this UI node
+#[derive(Component, Copy, Clone, PartialEq, Debug, Serialize, Deserialize, Reflect)]
+#[reflect_value(PartialEq, Serialize, Deserialize)]
+pub enum Layout {
+    /// Use the absolute position and size specified.
+    Position,
+    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm.
+    Flex(FlexLayout),
+}
```

Comment:
@alice-i-cecile @TimJentzsch : I think we need to make some changes here:
Previously we mapped `Flex` and `None` to their corresponding taffy types.
```rs
match value {
    Display::Flex => taffy::style::Display::Flex,
    Display::None => taffy::style::Display::None,
}
```
but with change we are now doing the following:
```rs
if let Layout::Flex(flex_layout) = ... // Layout using taffy::flex
else panic!()
```
And `Layout::Position` does really not what it claims it does.

### Solutions
A: We map `Layout::Position` to `taffy::Display::None`, and supply the the full `FlexLayout`-struct here as well (I assume taffy still needs the full data even if it does not use it?)
A.1: Same as A but rename `Layout`-enum to `Display` again

B: We remove the `Layout`-enum completely and say that FlexBox is the _only_ layouting you can currently use in Bevy. (Does anyone really use the `None` option?)

Reply:
To clarify:

Option `A` looks like this:
```
pub enum Layout {
    /// Use the absolute position and size specified.
    Position(FlexLayout), // <-- We now supply the whole style here
    /// Use the [Flexbox](https://cssreference.io/flexbox/) layout algorithm.
    Flex(FlexLayout),
}
```
with everything else being the same.

Reply:
Agreed on your analysis here.

Reply:
I'm a bit confused, was `Layout::Position` introduced in this PR?

If it's a non-flexbox layout algorithm I'd expect it to not use taffy at all.
`taffy::Display::None` would mean that no space is allocated for the element in the layout.
If we already have multiple layout algorithms I'm not sure how they'd mix together

Reply:
> I'm a bit confused, was `Layout::Position` introduced in this PR?

Yes. It was `Display::None` -> `Layout::None` -> `Layout::Position`, then after that `Layout::Flex` was made to contain the `FlexLayout` instead of the reverse (where `Style` contained `Display`)

> If it's a non-flexbox layout algorithm I'd expect it to not use taffy at all.
taffy::Display::None would mean that no space is allocated for the element in the layout.
If we already have multiple layout algorithms I'm not sure how they'd mix together

We don't have multiple layouting algorithms. We only have what taffy supplies so it would be false to create `Layout::Position` where there in fact is nothing. I think there's a series of doc-comments that have caused the confusion.

I vote we remove `Layout` as an enum and keep the `taffy::Display` in the `FlexLayout` struct so that users can "collapse" their elements if they wish.

Reply:
Yea I think we should keep the `Layout` struct for actual layout algorithms (or remove it since we have only one).

I was thinking if we could move this to the `Visibility` struct, but I think that's also used for other stuff so we can't do that.

Moving it to `FlexLayout` probably makes sense.

Reply:
FWIW, I'm anticipating `Display` in Taffy eventually becoming something more like a `TypeId`, such that we can accept any layout method that implements the requisite trait (which could include ones as simple as Swift-UI's `Padding` component, or as complex as a virtualised `TableView` widget).

I'm also considering the possibility that it might make sense to move responsibility for `Display` out of Taffy entirely, such that embedders of Taffy like bevy_ui supply a function which determines which display mode to use for a node and just calls into Taffy for the actual layout algorithm if it wants to use one that Taffy implements. Possibly we could have both options.

It moves away from the CSS model a bit, but it may make sense to have an enum like:

```
enum LayoutMode {
    /// Node should laid out as part of the main layouting algorithm in the parent. It may affect the layout of it's siblings.
    InFlow,
    /// Node should be positioned after the main layout algorithm has completed. It does not affect the layout of it's siblings.
    OutOfFlow,
    /// Node should not be 
    Hidden,
}
```
CSS spreads this out over the Display and Position properties, but I think that's more historical accident than anything. Morphorm has something like this [here](https://docs.rs/morphorm/0.3.0/morphorm/types/enum.PositionType.html), although it doesn't include the hidden variant).